src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemExtensionsTests.cs

```diff
@@ -52,26 +53,114 @@ public void SarifWorkItemExtensions_CreateWorkItemTitle_HandlesSingleResultWithR
             sb.Length.Should().Be(0, because: Environment.NewLine + sb.ToString());
         }
 
+        private class PhraseToolNamesTestCase
+        {
+            public PhraseToolNamesTestCase(List<string> input, string expectedOutput)
+            {
+                Input = input;
+                ExpectedOutput = expectedOutput;
+            }
+
+            public List<string> Input { get; }
+            public string ExpectedOutput { get; }
+        }
+
+        private static readonly ReadOnlyCollection<PhraseToolNamesTestCase> s_phraseToolNamesTestCases =
+            new ReadOnlyCollection<PhraseToolNamesTestCase>(new PhraseToolNamesTestCase[]
+            {
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {null},
+                    expectedOutput: "''"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {""},
+                    expectedOutput: "''"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"CredentialScanner"},
+                    expectedOutput: "'CredentialScanner'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"CredentialScanner", "Binskim"},
+                    expectedOutput: "'CredentialScanner' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"SEMMLE", "CredentialScanner", "Binskim"},
+                    expectedOutput: "'SEMMLE', 'CredentialScanner' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"SEMMLE", "", "Binskim"},
+                    expectedOutput: "'SEMMLE' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"", "CredentialScanner", "SEMMLE", "Binskim"},
+                    expectedOutput: "'CredentialScanner', 'SEMMLE' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"", "CredentialScanner"},
+                    expectedOutput: "'CredentialScanner'"),
+            });
```

Comment:
>} [](start = 12, length = 1)

Sorry I missed this the first time through: You should have test cases for a `null` list and for an empty list. `null` should probably throw (remember we discussed adding ShouldThrow to the test case class for this situation; your ctor can default it to `false`). Empty list should ... I dunno, up to you.